blueocean-web/src/main/resources/io/jenkins/blueocean/PageStatePreloadDecorator/header.jelly

```diff
@@ -0,0 +1,39 @@
+<j:jelly xmlns:j="jelly:core" xmlns:st="jelly:stapler">
+  <script>
+    (function () {
+      // construct the state object parent path inside window.$$blueocean.
+      function setState(statePropertyPath, state) {
+        var pathTokens = ('$blueocean.' + statePropertyPath).split('.');
+        var contextObj = window;
+
+        // Basically an Array shift
```

Comment:
@sophistifunk good question .... I don't remmber why I did that lol. The fact that I added the "Basically an Array shift" comment suggests that I was aware of the similarity. Maybe the function started out different to how it ended up, with the final version basically being the same as a shift.